Tools/scripts/size_compare_branches.py

```diff
@@ -57,6 +57,7 @@ def __init__(self,
                  all_boards=False,
                  use_merge_base=True,
                  waf_consistent_builds=True,
+                 show_empty=True,
```

Comment:
why do you set the default to false and the init to be true?

Reply:
Well spotted :-)

I did stare at this for a while.

The default invocation for the script I wanted to be true, that was for sure, and that was why it got that default.

The method's default was a bit different; if the method gets re-used then I thought the calling process might want all of the data, not just some of it.  In particular, this parameter currently filters within the actual csv-generation stage, but I think it's entirely possible that filtering could get moved out to somewhere else at some stage (if, for example, we wanted to support JSON output).  It would then make sense for more methods to appear which give calling processes access to the data in a rawer form - and this parameter would then affect those rawer-form methods.  Returning *all* the data from those rawer-form things seemed sensible.